Public/Src/Engine/Dll/EngineSchedule.cs

```diff
@@ -825,6 +826,11 @@ private bool ShouldSerializeOptimizationDataStructurePostExecution()
             }
         }
 
+        private static bool ShouldRemoveEmptyDirectories(IConfiguration configuration, string path)
+        {
+            return configuration.Schedule.UnsafeDisableSharedOpaqueEmptyDirectoryScrubbing && Directory.Exists(path) && !Directory.EnumerateFileSystemEntries(path).Any();

```

Comment:
>UnsafeDisableSharedOpaqueEmptyDirectoryScrubbing [](start = 42, length = 48)

Should it not be `!configuration.Scheduler.UnsafeDisableSharedOpaqueEmptyDirectoryScrubbing` ? It's a bit confusing because the empty directory is removed if the flag is disabled.

Reply:
If the flag is false, then false is returned. The caller will do an or, and since false is the identity for or, whether the path is part of the build or not will be unchanged by this.

If the flag is true, then we first check if it is a directory (it could be a file!), and then check if it is empty, so true will only be returned if it is a empty directory.